### DISCUSSION TITLE
Class types in AXAML?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#6919] [Q&A] [answered]
I need a little advice please.

I'm creating an XSD  for AXAML for use in the Visual Code extension "Xml Complete" (*). I want to capture all the class types which can be used in AXAML.

The author of the Xml Complete extension has already created an XSD for Avalonia, here:

https://github.com/rogalmic/vscode-xml-complete/blob/master/test/Avalonia/AvaloniaXamlSchema.Formatted.xsd

However, he includes only controls which are subtypes of `Avalonia.Controls.Control`. But this does not include, for example, such things as `DataTemplate`.

Q. Which Avalonia class names should go or NOT go into the XSD?

Here is what I'm using at the moment to select and build a list of types:

            foreach (var asmb in AppDomain.CurrentDomain.GetAssemblies())
            {
                if (asmb.FullName != null && asmb.FullName.StartsWith("Avalonia."))
                {
                    foreach (var type in asmb.GetExportedTypes())
                    {
                        // SELECTION HERE
                        if (type.IsClass && !type.IsAbstract && !type.IsGenericType &&
                            type.FullName?.StartsWith("Avalonia.") == true)
                        {
                            dest.Add(type);
                        }
                    }
                }
            }

Where I am referencing the following in my csproj file:

		<PackageReference Include="Avalonia" Version="0.*"/>
		<PackageReference Include="Avalonia.Desktop" Version="0.*"/>
		<PackageReference Include="Avalonia.Controls.DataGrid" Version="0.*"/>
		<PackageReference Include="Avalonia.ReactiveUI" Version="0.*"/>

Should I specifically include or exclude any namespaces? Or do something else? Any suggestions?

(*) https://github.com/rogalmic/vscode-xml-complete

**Comment by kekekeks** [comment, marked as answer]
Namespace mappings are defined via `XmlnsDefinitionAttribute`. See https://github.com/AvaloniaUI/Avalonia/search?q=XmlnsDefinition for the list of usages.

**Reply by kuiperzone** [reply]
That helps loads! Cheers
